canvasapi/communication_channel.py

```diff
@@ -66,11 +70,98 @@ def get_preference(self, notification):
         """
         response = self._requester.request(
             'GET',
-            'users/%s/communication_channels/%s/notification_preferences/%s' % (
+            'users/{}/communication_channels/{}/notification_preferences/{}'.format(
                 self.user_id,
                 self.id,
                 notification
             )
         )
         data = response.json()['notification_preferences'][0]
```

Comment:
See my comment below. We should have a discussion about the potential side effects of this line.